src/main/java/org/openhab/tools/analysis/checkstyle/PomXmlCheck.java

```diff
@@ -0,0 +1,245 @@
+            } else {
+                log(0, MISSING_PARENT_ARTIFACT_ID_MSG, file.getPath());
+            }
+        }
+
+        // get the version from the pom.xml
+        String versionNodeValue = getNodeValue(file, POM_VERSION_XPATH_EXPRESSION);
+        pomVersion = getVersion(versionNodeValue, pomVersionPattern);
+
+        // the version line will be preserved for finalization of the processing
+        String versionTagName = "version";
+        String versionLine = String.format("<%s>%s</%s>", versionTagName, versionNodeValue, versionTagName);
+        pomVersionLine = findLineNumber(lines, versionLine, 0);
+
+        // get the artifactId from the pom.xml
+        String artifactIdNodeValue = getNodeValue(file, POM_ARTIFACT_ID_XPATH_EXPRESSION);
+        pomArtifactId = artifactIdNodeValue;
+
+        // the artifact ID line will be used in the finalization as well
+        String artifactIdTagName = "artifactId";
+        String artifactIdLine = String.format("<%s>%s</%s>", artifactIdTagName, artifactIdNodeValue, artifactIdTagName);
+        pomArtifactIdLine = findLineNumber(lines, artifactIdLine, 0);
+    }
+
+    @Override
+    public void finishProcessing() {
+        compareProperties(pomVersion, manifestVersion, pomVersionLine, WRONG_VERSION_MSG, MISSING_VERSION_MSG);
+        compareProperties(pomArtifactId, manifestBundleSymbolicName, pomArtifactIdLine, WRONG_ARTIFACT_ID_MSG,
+                MISSING_ARTIFACT_ID_MSG);
+    }
+
+    private void compareProperties(String pomProperty, String manifestProperty, int wrongPropertyLine,
+            String wrongPropertyMessage, String missingPropertyMessage) {
+        if (pomProperty != null) {
+            if (manifestProperty != null && !pomProperty.equals(manifestProperty)) {
+                logMessage(pomDirectoryPath + File.separator + POM_XML_FILE_NAME, wrongPropertyLine, POM_XML_FILE_NAME,
+                        wrongPropertyMessage);
+            }
+            // no need to log if something is wrong with the manifestProperty,
+            // there are other special checks for that
+        } else {
+            logMessage(pomDirectoryPath + File.separator + POM_XML_FILE_NAME, 0, POM_XML_FILE_NAME,
+                    missingPropertyMessage);
+        }
+    }
+
+    private String getNodeValue(File file, String versionExpression) throws CheckstyleException {
+        Document xmlDocument = parseDomDocumentFromFile(file);
+
+        XPathExpression xPathExpression = compileXPathExpression(versionExpression);
+        Object result = null;
+        try {
+            result = xPathExpression.evaluate(xmlDocument, XPathConstants.NODESET);
+        } catch (XPathExpressionException e) {
+            logger.error("An exception was thrown, while trying to parse the file: {}", file.getPath(), e);
+        }
+
+        return getNodeValue(result);
+    }
+
+    private String getNodeValue(Object result) {
+        NodeList nodes = null;
+        if (result != null) {

```

Comment:
A lot of null checks which seem to be chained because you pass null in maybe limit passing around nulls